### PR TITLE
Dev > Main Talisman Update

### PR DESCRIPTION
**Pull request by rolandoo0**

### CODE REVIEW

File: server/__init__.py [hunk 1]
```diff
@@ -1,13 +1,22 @@
...
 
-app = Flask(__name__)
+csp = {
+    "default-src": "'self'",
+    "img-src": "*",
+    "script-src": "*",
+}
 
+app = Flask(__name__)
+Talisman(app, content_security_policy=csp)
```
Comment (adrian2x) [inline server/__init__.py]: use `content_security_policy=None`

File: server/models/user.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    def update(self, data: dict):
+        "Update a user document"
+        self.doc.update(data)
+        return self.doc.get()
+
+    def delete(self):
+        "Delete a user document"
+        self.doc.delete()
+        return {"message": "User deleted successfully."}
```
Comment (adrian2x) [inline server/models/user.py]: instead of returning a dict, return `self` instead.

File: server/models/post.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    def update(self, data: dict):
+        "Update a post document"
+        self.doc.update(data)
+        return self.doc.get()
+
+    def delete(self):
+        "Delete a post document"
+        self.doc.delete()
+        return {"message": "Post deleted successfully."}
```
Comment (adrian2x) [inline server/models/post.py]: instead of returning a dict, return `self` instead.

File: server/posts/posts.py [hunk 4]
```diff
@@ -20,30 +22,26 @@ def get_posts():
...
     return jsonify(updated_doc)
 
 
 @posts_blueprint.route("/<post_id>", methods=["DELETE"])
 def delete_post(post_id):
     "Delete post by post id"
-    doc = db.collection("posts").document(post_id)
-    doc.delete()
-    return jsonify({"message": "Post deleted successfully."})
+    return jsonify(Post(post_id).delete())
```
Comment (adrian2x) [inline server/posts/posts.py]: after deleting, return just `jsonify(ok=True)`

File: server/users/users.py [hunk 5]
```diff
@@ -36,24 +27,22 @@ def create_user():
...
+    updated_doc = User(uid).update(data).to_dict()
     return jsonify(updated_doc)
 
 
 @users_blueprint.route("/<uid>", methods=["DELETE"])
 def delete_user(uid):
     "Delete user by user id"
-    delete_firebase_user(uid)
+    User.delete_firebase_user(uid)
     return jsonify(success=True)
```
Comment (adrian2x) [inline server/users/users.py]: Return `jsonify(ok=True)`

File: server/models/user.py (hunk 2 above)
Comment (rolandoo0) [reply server/models/user.py]: why do we want to return `self` here

Comment (adrian2x) [reply server/models/user.py]: It is not strictly necessary but returning `self` from methods allows for this access pattern: `obj.foo().bar().baz()...`
Return that or None, but we don't need to return a dict.

**Pull request closed without merging**